tools/cr/post_presubmit.py

```diff
@@ -0,0 +1,159 @@
+
+    # Filtering out all comments that do not contain a presubmit hash as those
+    # are not relevant to our comment management.
+    existing_comments = [
+        comment for comment in existing_comments
+        if '<!-- presubmit-hash=' in comment["body"]
+    ]
+
+    active_report_hashes = set()
+    for category, notifications in presubmit_entries.items():
+        for report in notifications:
+            # Hashing of the report is done so we can add it to the comment
+            # itself as a html comment, and then check if it already exists,
+            # so we don't post the same comment multiple times.
+            report_hash = hashlib.sha256(
+                json.dumps(report,
+                           sort_keys=True).encode('utf-8')).hexdigest()
+            comment_hash = f'<!-- presubmit-hash={report_hash} -->'
+            active_report_hashes.add(report_hash)
+
+            if any(comment_hash in comment["body"]
+                   for comment in existing_comments):
+                print(
+                    f'Comment with hash {report_hash} already exists. Skipping.'
+                )
+                continue
+
+            def get_severity_message():
+                if category == 'errors':
+                    # There's code never gets here as a presubmit error causes
+                    # a CI failure, so the CI may end up never launching this
+                    # script. Also, in a few tests, it has not been very clear
+                    # whether or not PRESUBMIT generates JSON entries for
+                    # errors.
+                    return (
+                        '> [!CAUTION]\n'
+                        '> You have got a presubmit error. This will cause a '
+                        'CI failure.')
+                if category == 'warnings':
+                    return (
+                        '> [!WARNING]\n'
+                        '> You have got a presubmit warning. Please address '
+                        'it if possible.')
+                if category == 'notifications':
+                    return (
+                        '> [!NOTE]\n> You have got a presubmit notification. '
+                        'No action required.')
+
+                raise ValueError(f'Unknown category: {category}')
+
+            message = report.get('message', '')
+            body_content = get_severity_message()
+            if message:
+                body_content += f'\n\n```\n{message}\n```'
+
+            items = report.get('items', [])
+            if items:
+                body_content += '\n\nItems:\n```\n'
+                for item in items:
+                    body_content += f'{item}\n'
+                body_content += '```'
+
+            body_content += f'\n\n{comment_hash}'
+
+            subprocess.run([
+                'gh', 'api',
+                f'repos/brave/brave-core/issues/{pr_number}/comments', '-X',
+                'POST', '-F', 'body=@-'
```

Comment:
This may become a notification hell for reviewers. 

Do we need to check return code here and below (and do retries)?

Reply:
Maybe we should have some retry policies, although at the moment I want to have a trial run to see how this goes. I think we need to collect feedback from reviewers, because we can always tune it ourselves.

Regarding retries, I guess yes, we have to eventually define a policy of what we do if the comments fail to be posted. Do we retry? Do we eventually fail the CI? I think it may be better if we do this as a follow up maybe?